chaco/image_plot.py

```diff
@@ -372,3 +421,18 @@ def _array_bounds_from_screen_rect(self, image_rect):
         row_max = min(row_max, array_height)
```

Comment:
An edge case that isn't handled here occurs when `col_min` is greater than `array_width` or `col_max` is less than `0`. These only occur when the image disappears from the plot, e.g., panning the image to the right as much as possible and then zooming in leads to  negative `x_max` and `col_max`. 

While `x_min` and `col_min` are also negative in this example, `col_min` is explicitly clipped to 0, whereas `col_max` isn't, leading to `col_max - col_min < 0`.

When these bounds are used to compute the LOD in `_calculate_necessary_lod`:
https://github.com/enthought/chaco/blob/e41cfd375d611d3438328ced67fbe9a4c3182c39/chaco/image_plot.py#L432-L438
 no LOD will satisfy `col_max - col_min >= screen_rect[2]`, and so the necessary LOD is set to 0. This causes the highest resolution image to be unnecessarily cached, potentially crashing the program if the image is too large to fit in memory (actually, only the slice `[:, :-abs(col_max)]` is cached, but that can still be large).

The fix is simple:
```diff
-        col_min = max(col_min, 0)
-        col_max = min(col_max, array_width)
-        row_min = max(row_min, 0)
-        row_max = min(row_max, array_height)
+        col_min, col_max = np.clip([col_min, col_max], 0, array_width)
+        row_min, row_max = np.clip([row_min, row_max], 0, array_height)
```